File: lib/is-video.go
package lib

import "slices"

func IsVideo(s *string) bool {
	return slices.Contains([]string{
		".wmv",
		".asf",
		".asx",
		".rm",
		".rmvb",
		".mp4",
		".3gp",
		".mov",
		".m4v",
		".avi",
		".dat",
		".mkv",
		".flv",
		".vob",
		".mpg",
		".mpeg",
		".ts",
	}, *s)
}

File: web_src/lib/is-video.ts
export const isVideo = (extName: string): boolean => {
  const formatList = [
    ".wmv",
    ".asf",
    ".asx",
    ".rm",
    ".rmvb",
    ".mp4",
    ".3gp",
    ".mov",
    ".m4v",
    ".avi",
    ".dat",
    ".mkv",
    ".flv",
    ".vob",
    ".mpg",
    ".mpeg",
    ".ts",
  ];

  return formatList.includes(extName);
};
